src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractDatabaseEngine.java

```diff
@@ -415,7 +416,13 @@ public synchronized Connection getConnection() throws RetryLimitExceededExceptio
         }
     }
 
-    private synchronized void recover() throws DatabaseEngineException, NameAlreadyExistsException {
+    /**
+     * Clears entities and prepared statements cache.
+     *
+     * @throws DatabaseEngineException    If an error occurs.
+     * @throws NameAlreadyExistsException If an error occurs.
+     */
+    protected synchronized void recover() throws DatabaseEngineException, NameAlreadyExistsException {
```

Comment:
Had to change it to `protected` so it could be called in Oracle Engine and thus it could clear PDB statement cache and recreate all of them again.